subsys/bluetooth/host/adv.c

```diff
@@ -572,17 +581,26 @@ static int hci_set_adv_ext_fragmented(struct bt_le_ext_adv *adv, uint16_t hci_op
 	while (!ad_stream_is_empty(&stream)) {
 		struct bt_hci_cp_le_set_ext_adv_data *set_data;
 		struct net_buf *buf;
+		size_t data_len, cmd_size;
 		int err;
 
-		buf = bt_hci_cmd_create(hci_op, sizeof(*set_data));
+		(void)memcpy(&backup, &stream, sizeof(stream));
+
+		data_len = ad_stream_read(&stream, NULL, sizeof(set_data->data));
```

Comment:
I prefer that the copy `backup` is what is used up. It's more function-like then.

I think the stream copy and read must be a new API method `ad_stream_peek`. The fact `ad_stream` can be shallow-copied should be an implementation detail.